api/src/main/java/com/xwiki/googleapps/internal/GoogleAppsXWikiObjects.java

```diff
@@ -0,0 +1,608 @@
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.googleapps.internal;
+
+import java.io.File;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Provider;
+
+import org.apache.commons.httpclient.util.DateUtil;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.slf4j.Logger;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.model.reference.ObjectReference;
+import org.xwiki.observation.ObservationManager;
+import org.xwiki.query.Query;
+import org.xwiki.query.QueryManager;
+import org.xwiki.stability.Unstable;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.doc.XWikiAttachment;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xwiki.googleapps.DriveDocMetadata;
+import com.xwiki.googleapps.GoogleAppsException;
+
+class GoogleAppsXWikiObjects implements GoogleAppsConstants
+{
+    // environment
+    private final Logger log;
+
+    private final Provider<XWikiContext> contextProvider;
+
+    private final QueryManager queryManager;
+
+    private final DocumentReferenceResolver<String> documentResolver;
+
+    private final DocumentReferenceResolver<String> userResolver;
+
+    private final File permanentDir;
+
+    private boolean started;
+
+    private Boolean configActiveFlag;
+
+    private boolean useCookies;
+
+    private boolean skipLoginPage;
+
+    private boolean authWithCookies;
+
+    private String configAppName;
+
+    private String configClientId;
+
+    private String configClientSecret;
+
+    private String configDomain;
+
+    private boolean configScopeUseAvatar;
+
+    private boolean configScopeUseDrive;
+
+    private int configCookiesTTL;
+
+    // internal objects
+    private DocumentReference configDocRef;
+
+    private ObjectReference configObjRef;
+
+    private GoogleAppsEventListener eventListener;
+
+    private DocumentReference syncdocClassRef;
+
+    private DocumentReference gauthClassRef;
+
+    GoogleAppsXWikiObjects(Logger log, Provider<XWikiContext> contextProvider,
+            QueryManager queryManager, DocumentReferenceResolver<String> documentResolver,
+            DocumentReferenceResolver<String> userResolver, File permanentDir)
```

Comment:
Why isn't this a component? It feels so bad to have to pass all these dependencies through the constructor when you can simply inject them..